src/segment_reshape/topology/find_related.py

```diff
@@ -99,3 +79,235 @@ def calculate_common_segment(
 #   - jos nauhalle lisätessä on laitettu joku erikois-id lippu (IS_RESHAPE_TRAIL),
 #     voidaan käsitellä halutulla tavalla
 #   - tehdään muutokset kohteille
+
+
+def get_common_geometries(
+    main_feature_layer: QgsVectorLayer,
+    main_feature: QgsFeature,
+    related_features_by_layer: List[Tuple[QgsVectorLayer, QgsFeature]],
+    trigger_location: QgsPoint,
+) -> CommonGeometriesResult:
+
+    trigger_geom = QgsGeometry(trigger_location)
+    main_feature_geom = main_feature.geometry()
+
+    # Find geometries having their boundary intersecting trigger location
+    intersecting_geoms: List[QgsGeometry] = []
+    for _, feature in related_features_by_layer:
+        geom = feature.geometry()
+        if (
+            geom.type() == QgsWkbTypes.PolygonGeometry and geom.touches(trigger_geom)
```

Comment:
Even though mypy will not complain about this old-style enum access, if we target 3.22+ a more suitable method is to use the actual sub-class of the correct type, ie. `geom.type()` is typed to return `QgsWkbTypes.GeometryType`, and the check can be made with `== QgsWkbTypes.GeometryType.PolygonGeometry`